File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    count = 0
    heapq.heapify(scoville)
    while scoville:
        temp = heapq.heappop(scoville)
        if temp >= K:
            break
        elif not scoville and temp < K:
            return -1
        heapq.heappush(scoville, temp + heapq.heappop(scoville) * 2)
        count += 1
    return count

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(new Comparator<Integer>() {
            @Override
            public int compare(Integer num1, Integer num2) {
                return num1 - num2;
            }
        });
        
        for (int s : scoville)
            pq.offer(s);
        
        int count = 0;
        while (pq.size() > 1) {
            if (pq.peek() < K) {
                pq.offer(pq.poll() + pq.poll()*2);
                count++;
            }
            else
                break;
        }
        
        if (pq.size() == 1 && pq.peek() < K)
            return -1;
        return count;
    }
}